library/x509_crt.c

```diff
@@ -1184,8 +1184,17 @@ int mbedtls_x509_crt_parse_path( mbedtls_x509_crt *chain, const char *path )
 
     while( ( entry = readdir( dir ) ) != NULL )
     {
-        mbedtls_snprintf( entry_name, sizeof entry_name, "%s/%s", path, entry->d_name );
-        //entry_name length is MBEDTLS_X509_MAX_FILE_PATH_LEN  and it includes both path and filename
+        if( strlen(path) + strlen(entry->d_name) + 1 < MBEDTLS_X509_MAX_FILE_PATH_LEN )
+        {
+            mbedtls_snprintf( entry_name, sizeof entry_name, "%s/%s", path, entry->d_name );         
+        }
+        else
+        {
+            closedir( dir );
+            ret = MBEDTLS_ERR_X509_FILE_IO_ERROR;
```

Comment:
Please change the error in this case to `MBEDTLS_ERR_X509_BUFFER_TOO_SMALL`, to align the behavior with Mbed TLS ≥2.4.